features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp

```diff
@@ -0,0 +1,333 @@
+#include "ll_defs.h"
+#include "chci_drv.h"
+#include "lhci_api.h"
+#include "platform_api.h"
+#include "platform_ble_api.h"
+#include "wsf_assert.h"
+#include "wsf_buf.h"
+#include "wsf_timer.h"
+#include "wsf_trace.h"
+#include "dm_api.h"
+
+// Nordic Includes
+#include "nrf.h"
+
+#include "NRFCordioHCIDriver.h"
+#include "NRFCordioHCITransportDriver.h"
+
+#ifdef DEBUG
+#include <stdio.h>
+#include <stdarg.h>
+#define DBG_WARN(...) printf(__VA_ARGS__)
+#else
+#define DBG_WARN(...)
+#endif
+
+using namespace ble::vendor::nordic;
+using namespace ble::vendor::cordio;
+
+/*! \brief      Memory that should be reserved for the stack. */
+#define CORDIO_LL_MEMORY_FOOTPRINT  41906UL
+
+/*! \brief      Typical implementation revision number (LlRtCfg_t::implRev). */
+#define LL_IMPL_REV             0x2303
+
+// Note to implementer: this should be amended if the Cordio stack is updated
+
+// The Nordic-specific baseband configuration
+// The BB_ config macros are set in the bb_api.h header file
+const BbRtCfg_t NRFCordioHCIDriver::_bb_cfg = {
+    /*clkPpm*/ 20,
+    /*rfSetupDelayUsec*/ BB_RF_SETUP_DELAY_US,
+    /*maxScanPeriodMsec*/ BB_MAX_SCAN_PERIOD_MS,
+    /*schSetupDelayUsec*/ BB_SCH_SETUP_DELAY_US
+};
+
+static const uint16_t maxAdvReports = 16;
+static const uint16_t numRxBufs      = 8;
+#if !defined(NRF52840_XXAA)
+static const uint16_t advDataLen     = 128;
+static const uint16_t connDataLen    = 256;
+static const uint16_t numTxBufs      = 8;
+#else
+static const uint16_t advDataLen     = LL_MAX_ADV_DATA_LEN;
+static const uint16_t connDataLen    = 512;
+static const uint16_t numTxBufs      = 16;
+#endif
+
+
+/* +12 for message headroom, + 2 event header, +255 maximum parameter length. */
+static const uint16_t maxRptBufSize = 12 + 2 + 255;
+
+/* +12 for message headroom, +4 for header. */
+static const uint16_t aclBufSize = 12 + connDataLen + 4 + BB_DATA_PDU_TAILROOM;
+
+const LlRtCfg_t NRFCordioHCIDriver::_ll_cfg = {
+    /* Device */
+    /*compId*/         			LL_COMP_ID_ARM,
+    /*implRev*/					LL_IMPL_REV,
+    /*btVer*/         			LL_VER_BT_CORE_SPEC_5_0,
+    0, // padding 
+    /* Advertiser */
+    /*maxAdvSets*/         		4, // 4 Extended Advertising Sets
+    /*maxAdvReports*/         	8,
+    /*maxExtAdvDataLen*/         advDataLen,
+    /*defExtAdvDataFrag*/        64,
+    0, // Aux delay
+    /* Scanner */
+    /*maxScanReqRcvdEvt*/         4,
+    /*maxExtScanDataLen*/         advDataLen,
+    /* Connection */
+    /*maxConn*/          4,
+    /*numTxBufs*/          numTxBufs,
+    /*numRxBufs*/          numRxBufs,
+    /*maxAclLen*/          connDataLen,
+    /*defTxPwrLvl*/          0,
+    /*ceJitterUsec*/          0,
+    /* DTM */
+    /*dtmRxSyncMs*/          10000,
+    /* PHY */
+    /*phy2mSup*/          TRUE,
+    /*phyCodedSup*/          TRUE,
+    /*stableModIdxTxSup*/         TRUE,
+    /*stableModIdxRxSup*/          TRUE
+};
+
+extern "C" void TIMER0_IRQHandler(void);
+
+static void idle_hook(void)
```

Comment:
Do we need that function ?

Reply:
Prior to adding that hook we had multiple issues with sleep latency - let's try and investigate how we could make it nicer once this one is in

Reply:
Apparently the bail may never happen, because `nextExpiration` (`uint32_t`) cannot be greater than 0 and less than 1 at the same time? If this is indeed the case, this idle hook is the same as the default hook.